File: js/198_HouseRobber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    if(nums.length === 0) return 0;
    var t = []
    for( var i = 0; i < nums.length; ++i){
        if(i === 0) t[0] = nums[0];
        else if(i === 1) t[1] = Math.max(nums[1], nums[0]);
        else t[i%2] = Math.max( t[(i-2)%2]+nums[i], t[(i-1)%2]);
    }
    return t[(nums.length-1)%2];
};

File: cpp/198_HouseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        int t[2];
        for(int i = 0; i < nums.size(); i++){
            if(i == 0) t[0] = nums[0];
            else if(i == 1) t[1] = std::max(nums[0], nums[1]);
            else t[i%2] = std::max( t[(i-2)%2] + nums[i], t[(i-1)%2]);
        }
        return t[(nums.size()-1)%2];
    }
};